src/components/Header/Header.jsx

```diff
@@ -1,11 +1,20 @@
 import React from "react";
 import "./Header.css";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
 import { BsCart, BsHeart, BsPersonCircle } from "react-icons/bs";
-import { useUser } from "../../context";
+import { useUser, useProducts } from "../../context";
 
 export const Header = () => {
   const { user } = useUser();
+  const { dispatch } = useProducts();
+  const navigator = useNavigate();
+
+  const onSearch = (e) => {
+   if(window.location.pathname !== '/products')
```

Comment:
You could have used `useLocation()` hooks from react router.

Reply:
Thanks, didn't knew that exists.

Reply:
done